runtime/jvmti/jvmtiLocalVariable.c

```diff
@@ -290,57 +290,74 @@ jvmtiSetLocalDouble(jvmtiEnv* env,
 	char signature,
 	jboolean isSet,
 	jboolean getLocalInstance)
 {
-	J9JVMTIEnv * j9env = (J9JVMTIEnv *) env;
-	J9JavaVM * vm = j9env->vm;
-	jvmtiError rc;
-	J9VMThread * currentThread;
+	J9JVMTIEnv *j9env = (J9JVMTIEnv *)env;
+	J9JavaVM *vm = j9env->vm;
+	jvmtiError rc = JVMTI_ERROR_NONE;
+	J9VMThread *currentThread = NULL;
 
 	rc = getCurrentVMThread(vm, &currentThread);
-	if (rc == JVMTI_ERROR_NONE) {
+	if (JVMTI_ERROR_NONE == rc) {
 		J9VMThread *targetThread = NULL;
 
 		vm->internalVMFunctions->internalEnterVMFromJNI(currentThread);
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
-			J9StackWalkState walkState;
-			UDATA objectFetched = FALSE;
+			J9StackWalkState walkState = {0};
+			BOOLEAN objectFetched = FALSE;
+			J9VMThread *threadToWalk = targetThread;
+
+#if JAVA_SPEC_VERSION >= 19
+			J9VMThread stackThread = {0};
+			J9VMEntryLocalStorage els = {0};
+			j9object_t threadObject = (NULL == thread) ? currentThread->threadObject : J9_JNI_UNWRAP_REFERENCE(thread);
+			J9VMContinuation *continuation = getJ9VMContinuationToWalk(currentThread, targetThread, threadObject);
+			if (NULL != continuation) {
+				vm->internalVMFunctions->copyFieldsFromContinuation(currentThread, &stackThread, &els, continuation);
+				threadToWalk = &stackThread;
+			}
+#endif /* JAVA_SPEC_VERSION >= 19 */
+
+#if JAVA_SPEC_VERSION >= 19
+			if (NULL != targetThread)
+#endif /* JAVA_SPEC_VERSION >= 19 */
+			{
+				vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+			}
 
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-			rc = findDecompileInfo(currentThread, targetThread, (UDATA)depth, &walkState);
+			rc = findDecompileInfo(currentThread, threadToWalk, (UDATA)depth, &walkState);
 			if (JVMTI_ERROR_NONE == rc) {
-				UDATA validateRC;
-				UDATA slotValid = TRUE;
-				UDATA * slotAddress;
+				UDATA validateRC = 0;
+				BOOLEAN slotValid = TRUE;
+				UDATA *slotAddress = NULL;
 				J9Method *ramMethod = walkState.userData3;
 				U_32 offsetPC = (U_32)(UDATA)walkState.userData4;
-				J9ROMMethod * romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(ramMethod);
+				J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(ramMethod);
 
 				if (getLocalInstance) {
-					if (romMethod->modifiers & J9AccStatic) {
+					if (OMR_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccStatic)) {
 						/* GetLocalInstance illegal on static methods */
 						validateRC = J9_SLOT_VALIDATE_ERROR_INVALID_SLOT;
-					} else if (romMethod->modifiers & J9AccNative) {
+					} else if (OMR_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccNative)) {
 						/* GetLocalInstance legal on non-static native methods */
 						validateRC = J9_SLOT_VALIDATE_ERROR_NONE;
 					} else {
 						/* GetLocalInstance == GetLocalObject for non-static, non-native methods */
 						validateRC = validateLocalSlot(currentThread, ramMethod, offsetPC, (U_32) slot, signature, TRUE);
 					}
-				} else {
```

Comment:
what is the reason for removing this `else`?

Reply:
Also, please re-add the `else`; it seems to have been accidentally removed.